gh-90556: add SharedMemory.rename on multiprocessing.shared_memory (#132130)
python/cpython: Doc/library/multiprocessing.shared_memory.rst

      This method has no effect on Windows, where the only way to delete a
      shared memory block is to close all handles.

>    .. method:: rename(newname, flags)

Reviewer: ```suggestion
   .. method:: rename(newname, flags=0)
```

>       Renames the underlying shared memory block to *newname*.

Reviewer: "Renames" is present tense, we want future tense:
```suggestion
      Rename the underlying shared memory block to *newname*.
```

      By default, if *newname* already exists, it will be unlinked beforehand.

>       Passing the :data:`!os.SHM_RENAME_EXCHANGE` as *flags*,
>       an exchange with *newname* will be proceeded instead.
>       Passing the :data:`!os.SHM_RENAME_NOREPLACE` as *flags*,
>       an error will be triggered if *newname* already exists.

Reviewer: I don't really like how this reads. Here's how I would do it, but feel free to make any changes you want:
```suggestion

      *flags* (0 by default) takes :ref:`bitwise ORed <bitwise>` flags together.
      
      * :data:`!os.SHM_RENAME_EXCHANGE` will let
         an exchange with *newname* will be processed instead.
      * :data:`!os.SHM_RENAME_NOREPLACE` will raise
         an error will be triggered if *newname* already exists.

```

Ideally, we should document `SHM_RENAME_EXCHANGE` and `SHM_RENAME_NOREPLACE` with links to this though.

Author: Hi, I moved these constants from `os` to `multiprocessing.shared_memory` and added documents on them.
